make clean in the geopm root directory deletes the 'man' symlink

**Describe the bug**
Running make clean in the geopm root directory deletes the 'man' symlink in that directory.

Reproducer:
$> cd <geopm root directory>
$> ./autogen.sh
$> ./configure
$> ls -la man
lrwxrwxrwx 1 sidjana users 22 May 30 16:36 man -> service/docs/build/man
$> make clean
$> ls -la man
ls: cannot access 'man': No such file or directory
$> make --dry-run
.....
make[1]: *** No rule to make target 'man/GEOPM_CXX_MAN_Agent.3', needed by 'all-am'.  Stop.
make[1]: Leaving directory '/home/sidjana/projects/backup/geopm'
make: *** [Makefile:3032: all] Error 2

--- a/Makefile.am
+++ b/Makefile.am
@@ -573,7 +573,6 @@ DISTCLEANFILES = VERSION MANIFEST
 CLEAN_LOCAL_TARGETS= clean-local-coverage \
                      clean-local-fortran \
                      clean-local-googletest \
-                     clean-local-man \
                      # end
 
 clean-local: $(CLEAN_LOCAL_TARGETS)
@@ -582,9 +581,6 @@ distclean-local: distclean-local-ruby
 
 clean-local-coverage:
 	rm -f `find -name \*.gcno` `find -name \*.gcda`
-
-clean-local-man:
-	rm -rf man
 
 clean-local-fortran:
 	rm -rf geopm_prof.mod
@@ -643,7 +639,6 @@ TESTS = test_license.sh
 
 PHONY_TARGETS = clean-local \
                 clean-local-coverage \
-                clean-local-man \
                 distclean-local \
                 gh_pages \
                 rpm \

--- a/service/docs/source/index.rst
+++ b/service/docs/source/index.rst
@@ -5,7 +5,7 @@ Welcome to GEOPM
 
 
 The Global Extensible Open Power Manager (GEOPM) is a framework for exploring
-power and energy optimizations targeting heterogenous platforms. The GEOPM
+power and energy optimizations targeting heterogeneous platforms. The GEOPM
 package provides many built-in features. A simple use case is reading hardware
 counters and setting hardware controls with platform independent syntax using
 a command line tool on a particular compute node. An advanced use case is

--- a/service/dox/blurb.md
+++ b/service/dox/blurb.md
@@ -14,5 +14,5 @@ GEOPM Systemd Service and Daemon
 * A high performance interface enabling batch access to validated sets
   of signals/controls.
 
-* Supports extensibility for heterogenous environments through C++
+* Supports extensibility for heterogeneous environments through C++
   plugin infrastructure (IOGroups).

